### PR TITLE
Allow injection of io.Writer to each Logger implementation. Add tests for logger package.

### PR DESCRIPTION
**Pull request by metmajer**
This PR supports the injection of a `type io.Writer` to each logger function via the variadic parameter `a ...interface{}` in [cutil/logger/logger.go](https://github.com/kris-nova/kubicorn/blob/master/cutil/logger/logger.go). The added functionality is used in the accompanying unit tests, and also supports testing of higher level functionality that makes use of logging, as described in https://github.com/kris-nova/kubicorn/pull/194.

### CODE REVIEW

File: cutil/logger/logger_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+const (
+	format = "%v, %v, %v, all eyes on me!"
```
Comment (kris-nova) [inline cutil/logger/logger_test.go]: Did you just quote tupac?

Comment (metmajer) [inline cutil/logger/logger_test.go]: Feel free to choose a different song... :P

Comment (grillz) [inline cutil/logger/logger_test.go]: excellent work

Comment (kris-nova) [inline cutil/logger/logger_test.go]: This is the best pull request I have seen in my life

Comment (xmudrii) [inline cutil/logger/logger_test.go]: This made me laugh a lot :joy:

Comment (metmajer) [inline cutil/logger/logger_test.go]: Lol :-) Thanks all for your feedback!

**Pull request merged**